liota/core/device_discovery.py

```diff
@@ -471,36 +481,38 @@ def device_msg_process(self, data):
-            if key_dcc is None:
-                continue
-            if len(key_dcc) is 0:
-                continue
-            unique_key = type_key_map.get(key)
-            if unique_key is None:
-                continue
-            unique_key_value = ''
-            for k, v in value.iteritems():
-                if k == unique_key:
-                    unique_key_value = v
-            name = key + '_' + unique_key_value;
-            self._save_devinfo(name, key)
-            for dcc in key_dcc[:]:
-                # register device to dcc, set properties for the device
-                if 'iotcc' in dcc.lower():
-                    prop_dict = self.add_organization_properties("iotcc", value)
-                else:
-                    prop_dict = value
-                # TBM: temporarily assume dcc pkg will register edge_system as dcc name + "_edge_system"
-                edge_system_pkg = dcc + str("_edge_system")
-                (dev, reg_dev) = self.reg_device(dcc, edge_system_pkg, name, key, prop_dict)
-                if (dev is not None) and (reg_dev is not None):
-                    reg_rec = {}
-                    # add it in Records: currently only for debugging
-                    reg_rec[dcc] = (dev, reg_dev)
-                    self._update_devinfo(name, reg_rec)
-
+        try:
+            for key, value in data.iteritems():
+                key_dcc = type_dcc_map.get(key)
+                if key_dcc is None:
+                    continue
+                if len(key_dcc) is 0:
+                    continue
+                unique_key = type_key_map.get(key)
+                if unique_key is None:
+                    continue
+                unique_key_value = ''
+                for k, v in value.iteritems():
+                    if k == unique_key:
+                        unique_key_value = v
+                name = key + '_' + unique_key_value;
+                self._save_devinfo(name, key)
+                for dcc in key_dcc[:]:
+                    # register device to dcc, set properties for the device
+                    if 'iotcc' in dcc.lower():
+                        prop_dict = self.add_organization_properties("iotcc", value)
+                    else:
+                        prop_dict = value
+                    # TBM: temporarily assume dcc pkg will register edge_system as dcc name + "_edge_system"
+                    edge_system_pkg = dcc + str("_edge_system")
+                    (dev, reg_dev) = self.reg_device(dcc, edge_system_pkg, name, key, prop_dict)
+                    if (dev is not None) and (reg_dev is not None):
+                        reg_rec = {}
+                        # add it in Records: currently only for debugging
+                        reg_rec[dcc] = (dev, reg_dev)
+                        self._update_devinfo(name, reg_rec)
+        except:
+            log.exception("device_msg_process error:{0}".format(sys.exc_info()[0]))
```

Comment:
I re-thought it over, and I have to change it back to log.error(""). An exception when process a received device message is expected (it may occur), and we just need to report it, and device discovery thread can still listen and process more received messages.